models/user.rb

```diff
@@ -42,7 +42,7 @@ def unfollow(other_user)
   end
 
   def build_timeline()
-    if $redis.exists("timeline:user:#{self.id}")
+    if $redis.ttl("timeline:user:#{self.id}")>0
```

Comment:
`ttl` gets remaining time to expire, `exists` gets whether key exists. 
My guess is `exists` is faster since it returns boolean directly and then no need to compute next inequality.